Client sync events

adam-26 opened the pull request:
Add standard backbone sync events to client proxy. Currently, these events are not triggered on the client.

I attempted to include some unit tests, however sinonjs fake XmlHttpRequests feature (http://sinonjs.org/docs/#server) does not play nicely with requirejs.

pknepper commented:
I'm not sure this is necessary.  The default backbone sync only triggers a request event.  Other events should be handled by the default backbone fetch, save, destroy methods which delegate to the sync.

adam-26 commented:
You're right. Do you think the events triggered by the callSyncher should have different names from the standard backbone event names?

pknepper commented:
Good question.  I'm thinking the callSyncher should have different names to semantically differentiate itself from the standard backbone events.

adam-26 commented:
I agree. I've reverted the sync event triggers and changed the names of the call events. I wasn't really sure exactly what to call them. If you change the names of the events when you merge the PR, can you please post the event names here? :)
